dvc/repo/update.py

```diff
@@ -3,11 +3,11 @@
 
 @locked
 def update(self, target, rev=None):
-    from dvc.stage import Stage
+    from ..dvcfile import Dvcfile
 
-    stage = Stage.load(self, target)
+    stage = Dvcfile(self, target).load()
     stage.update(rev)
 
-    stage.dump()
+    stage.dvcfile.dump(stage)
```

Comment:
Feels like it is still backwards. In your upcoming PR dvcfile will take the central place, so why not do it justice and use properly:
```
dvcfile = Dvcfile(self, target)
dvcfile.stage.update()
dvcfile.dump()
```

We've discussed `stage.dvcfile.dump()` more as a hack for old tests to avoid messing around with them too much, but the core code should handle this properly right away.

Reply:
For the multistage, it makes more sense to have `dvcfile.dump(stage)`. At the moment, Dvcfile only have a single stage looks like a special case to me (and, hence I am generalizing).

Reply:
I agree, but it is a good opportunity to get ready for that. Even for single-stage, the logic is that Dvcfile writes and reads stuff, not the stage, so we need to use it properly right away. `stage.dvcfile.dump(stage)` is extremely ugly for the core code.

Reply:
@skshetry Or something like
```
dvcfile = Dvcfile(self, target)
stage = dvcfile.load()
stage.update()
dvcfile.dump(stage)
```
in that case. Just want to get the logic straight right away.

Reply:
> @skshetry Or something like

Yup, I went with something like that.

Reply:
But, writing `dvcfile = Dvcfile(self, target)` is still cumbersome.